src/http/server.rs

```diff
@@ -1,4 +1,31 @@
 //! HTTP server
+//!
+//! Provides a HTTP(S) server in `EspHttpServer`, plus all related structs.
+//!
+//! Typical usage of `EspHttpServer` involves creating a function (or closure)
+//! for every URI+method that the server is meant to handle. A minimal server that
+//! only handles HTTP GET requests to `index.html` looks like this:
+//!
+//! ```
+//! use esp_idf_svc::http::server::{Configuration, EspHttpServer};
+//!
+//! let mut server = EspHttpServer::new(&Configuration::default())?;
+//!
+//! server.fn_handler("/index.html", Method::Get, |request| {
+//! 	request.into_ok_response()?
```

Comment:
This example would no longer compile since today. You either need to annotate the exact error type you are returning using a turbofish syntax on the `Ok(())` or you should change this code to just be 
```
request.into_ok_response()?
    .write_all("<html><body>Hello world!</body></html>".as_bytes())
```

... without the final `Ok`
```

Reply:
Changed to "If you want to keep the server running indefinitely then make sure it's not dropped - you may add an infinite loop after the server is created, use `core::mem::forget`, or keep around a reference to it somehow.".

(Doesn't show as a change since it's a couple of lines down)

Reply:
Changed. (Doesn't show as a change since it's a couple of lines down)

Reply:
Argh, this is now even worse and not what I suggested. Np, I'll fix post-commit.